r/Dockerfile

```diff
@@ -73,14 +64,16 @@ ENV RENV_LIB=/src/lib
 RUN echo ".libPaths(c('$RENV_LIB', .libPaths()))" >> $(R RHOME)/etc/Rprofile.site
 
 # install runtime system deps
+# python3-dev prevents 'reticulate can only bind to copies of Python built with --enable-shared'
 # cleanup setup files
-RUN echo "python3-pip" >> sysdeps_run.txt && \
+RUN echo "python3-pip python3-venv python3-dev" >> sysdeps_run.txt && \
     cat sysdeps_run.txt | xargs ./install_debian_packages.sh && \
     rm -rf *
 
-# link system libstdc++ to conda installed version
-RUN rm /usr/lib/x86_64-linux-gnu/libstdc++.so.6 && \
-    ln -s /src/r-miniconda/envs/r-reticulate/lib/libstdc++.so.6 /usr/lib/x86_64-linux-gnu/libstdc++.so.6
+# install umap-learn for RunUMAP
+ENV WORKON_HOME=/src/.virtualenvs
+RUN R -q -e "reticulate::virtualenv_create('r-reticulate')" && \
+    R -q -e "reticulate::virtualenv_install('r-reticulate', 'umap-learn==0.5.3', pip_options='--no-cache-dir')" 
```

Comment:
motivation is reducing image size (before: ~1010 MB, after: 884MB) which speeds up worker pod being ready.

Together with python image size reduction (~530MB) worker ready time is down from ~2:30 --> 1:40

Reply:
finally removing conda! I tried something similar months ago but got stuck. We can port this to the pipeline as well